### ISSUE TITLE
Implement MIPS3 target support

### ISSUE THREAD

**Issue by KiritoDv** [c3lang/c3c#1392]
It would be nice to have mips-3 support for some homebrew platforms like n64 or some modems.

Im not sure if this could be a good reference: https://github.com/llvm/llvm-project/issues/18632 - https://clang.llvm.org/doxygen/Driver_2ToolChains_2Arch_2Mips_8cpp_source.html

**Comment by lerno** [comment]
How would we test it?

**Comment by KiritoDv** [comment]
Thats a good question, i can setup a few n64 and router firmware tests or an example program

**Comment by lerno** [comment]
I will probably want to test it from MacOS, are there any easy helpers to do that?

**Comment by KiritoDv** [comment]
Wdym by helper 0:?

**Comment by lerno** [comment]
Software that will help you run and emulate MIPS3 code on MacOS.

**Comment by KiritoDv** [comment]
Ohh, you could try ares which is an n64 emulator or an irix emulator https://forums.irixnet.org/thread-1012-post-7165.html

**Comment by lerno** [comment]
Looks non-trivial. Is there a way to bring this to CI?

**Comment by KiritoDv** [comment]
Yeah i think, my other idea was to use something like this https://github.com/gioannides/MIPS_SIMULATOR/tree/master, seems to be smaller than using a full emulator
